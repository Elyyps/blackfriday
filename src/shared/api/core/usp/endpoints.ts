import { uniqueSellingPoints } from "./generate-dummy-data";
import { USP } from "./usp";

export const generateUSP = (): USP => uniqueSellingPoints;
