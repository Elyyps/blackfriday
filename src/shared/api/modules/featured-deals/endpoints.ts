import { dummyFeaturedDealsData } from "./generate-dummy-data";

export const getFeaturedDealsDummyData = () => dummyFeaturedDealsData;
