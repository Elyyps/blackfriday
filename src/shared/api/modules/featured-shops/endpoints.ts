import { dummyFeaturedShopsData } from "./generate-dummy-data";

export const getFeaturedShopsDummyData = () => dummyFeaturedShopsData;
