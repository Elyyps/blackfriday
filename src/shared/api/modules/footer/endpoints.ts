import { bottomFooter, middleFooter, topFooter } from "./generate-dummy-data";
import { Footer } from "./models/footer.module";

export const getFullfooter = (): Footer => ({
  bottomFooter: bottomFooter(),
  middleFooter: middleFooter(),
  topFooter: topFooter()
});
