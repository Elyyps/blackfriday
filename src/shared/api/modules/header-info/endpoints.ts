import { HeaderInfoDummy } from "./generate-dummy-data";

export const getHeaderInfoDummyData = () => HeaderInfoDummy;
