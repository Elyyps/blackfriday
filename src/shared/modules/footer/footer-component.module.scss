.footer {
  font-size: 13px;
  font-weight: 600;
  color: $gray-light;
  background: $global-background;

  &__top {
    padding: 20px 0;
  }

  &__top-holder {
    justify-content: center;
    display: flex;
    align-items: center;
    flex-wrap: nowrap;
    @media (min-width: $breakpoint-medium) {
      justify-content: space-between;
    }
  }

  &__top-list {
    display: flex;
    font-size: 15px;
    color: $gray-light;
    margin: 0;
    @media (min-width: $breakpoint-medium) {
      margin-left: -15px;
    }
    li {
      padding: 0 16px;
      display: flex;
      align-items: center;
    }
    .icon-svg {
      margin-right: 10px;
    }
  }

  &__holder {
    background: $gray-dark;
    color: $global-inverse-color;
    padding: 48px 0 37px;
    @media (max-width: $breakpoint-xsmall-max) {
      padding: 16px 0;
    }
    a {
      text-decoration: none;
      color: $tint-gray;
      &:hover {
        color: $gray-lighter;
      }
    }
  }

  &__columns {
    @media (min-width: $breakpoint-small) {
      display: flex;
      flex-wrap: wrap;
    }
    @media (min-width: $breakpoint-medium) {
      flex-wrap: nowrap;
    }
    > div {
      @media (min-width: $breakpoint-small) {
        flex: 1 1 50%;
        width: 50%;
        padding-right: 32px;
        padding-bottom: 16px;
        display: flex;
        flex-direction: column;
        box-sizing: border-box;
      }
      @media (min-width: $breakpoint-medium) {
        flex: 1 1 auto;
        padding-bottom: 0;
      }
    }
  }

  &__column-body {
    flex: 1 1 auto;
    > p {
      margin-top: 0;
      margin-bottom: 15px;
    }
  }

  &__column-footer {
    flex: 0 0 auto;
    padding-top: 24px;
  }

  &__column-collapse {
    @media (max-width: $breakpoint-xsmall-max) {
      border-bottom: 1px solid $gray;
    }
  }
  &__column-desktop__container {
    display: block;
    @media (max-width: $breakpoint-xsmall-max) {
      display: none;
    }
  }
  &__column-mobile__container {
    display: none;
    @media (max-width: $breakpoint-xsmall-max) {
      display: block;
    }
  }
  &__column-title {
    font-size: 14px;
    font-family: $global-font-family-secondary;
    margin-bottom: 24px;
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: space-between;
    &__clicked {
      padding-bottom: 0 !important;
      color: $gray-lighter;
      .footer__column-title__icon {
        path {
          stroke: $gray-lighter;
        }
      }
    }
    &:hover {
      color: $gray-lighter;
      .footer__column-title__icon {
        path {
          stroke: $gray-lighter;
        }
      }
    }

    &__icon {
      margin-left: 15px;
      path {
        stroke: #fff;
      }
    }
    @media (max-width: $breakpoint-xsmall-max) {
      padding: 14px 0;
      position: relative;
      margin: 0;
    }
  }

  &__column-nav {
    margin: 0;

    > li {
      position: relative;
      list-style: none;
      padding-left: 16px;
      margin-bottom: 8px;
    }
    li:hover {
      a {
        color: white;
      }
    }
  }

  &__bottom {
    padding: 15px 0;
  }

  &__bottom-holder {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    @media (min-width: $breakpoint-small) {
      justify-content: space-between;
      align-items: center;
      flex-wrap: nowrap;
    }
  }

  &__logo-box {
    display: flex;
    align-items: center;
  }

  &__logo {
    width: 149px;
    display: block;
    margin-right: 32px;
  }

  &__nav {
    display: flex;
    flex-wrap: wrap;
    margin: 0 -16px 0 0;
    justify-content: flex-end;
    li {
      margin: 0 16px;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      justify-content: center;
      clear: both;
      width: 100%;
      margin: 0;
      order: -1;
      li {
        margin-bottom: 16px;
      }
    }

    a {
      text-decoration: none;
      color: $gray-dark;
      &:hover {
        color: #999;
      }
    }
  }

  &__social {
    &-items {
      padding: 8px;
    }
  }

  &__social-link {
    width: 100%;
    display: inline-flex;
    align-items: center;
  }

  &__social-icon {
    width: 36px;
    height: 36px;
    border-radius: $global-border-radius;
    border: solid 2px $gray;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-right: 16px;
    flex: 0 0 auto;
    path {
      fill: currentColor;
    }
  }

  &__social-text {
    font-size: 12px;
    color: $tint-gray;
  }
  &__social-title {
    font-size: 15px;
    color: $global-inverse-color;
  }
}

.rating {
  display: flex;
  align-items: center;
  font-size: 12px;
  @media (max-width: $breakpoint-small-max) {
    justify-content: center;
  }
  &__mark {
    font-size: 14px;
    margin-right: 8px;
    font-family: $global-font-family-secondary;
  }
}
