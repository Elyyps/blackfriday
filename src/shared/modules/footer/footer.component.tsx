import * as React from "react";
import { Link } from "react-router-dom";
import styles from "./footer-component.module.scss";
import { ImageComponent, Button, IconComponent, USP } from "@app/core/";

import { Footer } from "@app/api/modules/footer/models/footer.module";
import { LinkComponent } from "@app/core/link";
import { StarsRatingComponent } from "@app/core/stars-rating";
import { ExpandablePanelComponent } from "@app/core/expandable-panel";

export interface IFooterComponentProps {
  footerModule: Footer;
}
const CLOSE_PANEL = "close";
const numberStars = 5;
const FooterComponent = (props: IFooterComponentProps) => {
  const [openedPanel, setOpenedPanel] = React.useState<string | undefined>(undefined);

  const panelClicked = (key?: string) => {
    if (openedPanel === key) {
      setOpenedPanel(CLOSE_PANEL);
    } else {
      setOpenedPanel(key);
    }
  };

  const mainContent = () => (
    <div className={styles["footer__column-title"]}>{props.footerModule.middleFooter.socialMedia.title}</div>
  );

  return (
    <div className={styles["footer"]}>
      <div className={styles["footer__top"]}>
        <div className="uk-container">
          <div className={styles["footer__top-holder"]}>
            <div className={styles["footer__top-left"]}>
              <ul className={styles["footer__top-list"]}>
                {props.footerModule.topFooter.topLeftFooter.map((item, index) => (
                  <li key={index}>
                    <IconComponent icon={item.icon} size="20px" />
                    <span style={{ marginLeft: "10px" }}>{item.text}</span>
                  </li>
                ))}
              </ul>
            </div>
            <div className={`${styles["footer__top-right"]} uk-visible@m`}>
              <USP uspModule={props.footerModule.topFooter.usp} />
            </div>
          </div>
        </div>
      </div>
      <div className={styles["footer__holder"]}>
        <div className="uk-container">
          <div className={styles["footer__columns"]}>
            {props.footerModule.middleFooter.links.map((item, index) => (
              <div key={index} className={styles["footer__column-collapse"]}>
                <div className={styles["footer__column-desktop__container"]}>
                  <div className={styles["footer__column-title"]}>{item.title}</div>
                  <div className={styles["footer__column-collapse-body"]}>
                    <ul className={styles["footer__column-nav"]}>
                      {item.links.map((link, i) => (
                        <li key={i}>
                          <LinkComponent to={link.url}>{link.title}</LinkComponent>
                        </li>
                      ))}
                    </ul>
                  </div>
                </div>
                <div className={styles["footer__column-mobile__container"]}>
                  <ExpandablePanelComponent
                    callback={panelClicked}
                    identifier={item.id}
                    opened={openedPanel === item.id}
                    key={index}
                    mainContent={<div className={styles["footer__column-title"]}>{item.title}</div>}
                  >
                    <div>
                      <ul className={styles["footer__column-nav"]}>
                        {item.links.map((link, i) => (
                          <li key={i}>
                            <LinkComponent to={link.url}>{link.title}</LinkComponent>
                          </li>
                        ))}
                      </ul>
                    </div>
                  </ExpandablePanelComponent>
                </div>
              </div>
            ))}

            <div className={styles["footer__column-collapse"]}>
              <div className={styles["footer__column-desktop__container"]}>
                <div className={styles["footer__column-title"]}>
                  {props.footerModule.middleFooter.socialMedia.title}
                </div>
                <div className={styles["footer__column-collapse-body"]}>
                  <ul className={styles["footer__social"]}>
                    {props.footerModule.middleFooter.socialMedia.socialItem.map((item, index) => (
                      <li key={index}>
                        <a href={item.link} className={styles["footer__social-link"]} target="_blank">
                          <div className={styles["footer__social-icon"]}>
                            <IconComponent icon={item.image} size="15px" />
                          </div>
                          <div className={styles["footer__social-text"]}>
                            <div className={styles["footer__social-title"]}>{item.nameSocialMedia}</div>
                            {item.text}
                          </div>
                        </a>
                      </li>
                    ))}
                  </ul>
                </div>
              </div>

              <div className={styles["footer__column-mobile__container"]}>
                <ExpandablePanelComponent
                  callback={panelClicked}
                  identifier={props.footerModule.middleFooter.socialMedia.id}
                  opened={openedPanel === props.footerModule.middleFooter.socialMedia.id}
                  mainContent={mainContent()}
                >
                  <div className={styles["footer__social"]}>
                    {props.footerModule.middleFooter.socialMedia.socialItem.map((item, index) => (
                      <li key={index}>
                        <Link to={item.link} className={styles["footer__social-link"]}>
                          <div className={styles["footer__social-icon"]}>
                            <IconComponent icon={item.image} size="15px" />
                          </div>
                          <div className={styles["footer__social-text"]}>
                            <div className={styles["footer__social-title"]}>{item.nameSocialMedia}</div>
                            {item.text}
                          </div>
                        </Link>
                      </li>
                    ))}
                  </div>
                </ExpandablePanelComponent>
              </div>
            </div>
            <div>
              <div className={styles["footer__column-body"]}>
                <div className={styles["footer__column-title"]}>{props.footerModule.middleFooter.newsletter.title}</div>
                <p> {props.footerModule.middleFooter.newsletter.text}</p>

                <Button
                  title={props.footerModule.middleFooter.newsletter.buttonText}
                  icon={props.footerModule.middleFooter.newsletter.buttonIcon}
                  variant={"primary-inverted"}
                />
              </div>
              <div className={styles["footer__column-footer"]}>
                <div className={styles["rating"]}>
                  <div className={styles["rating__mark"]}>
                    {props.footerModule.middleFooter.googleReviews.rating > numberStars
                      ? 0
                      : props.footerModule.middleFooter.googleReviews.rating}
                  </div>
                  <StarsRatingComponent
                    rating={props.footerModule.middleFooter.googleReviews.rating}
                    // tslint:disable-next-line
                    voteTrigger={() => {}}
                    numberOfStars={numberStars}
                  />
                  <div style={{ marginLeft: "10px" }}>
                    <LinkComponent to={props.footerModule.middleFooter.googleReviews.link.url}>
                      {props.footerModule.middleFooter.googleReviews.link.title}
                    </LinkComponent>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className={styles["footer__bottom"]}>
        <div className="uk-container">
          <div className={styles["footer__bottom-holder"]}>
            <div className={styles["footer__logo-box"]}>
              <Link to="./" className={`uk-visible@m ${styles["footer__logo"]}`}>
                <ImageComponent src={props.footerModule.bottomFooter.logo} alt="image" />
              </Link>
              {props.footerModule.bottomFooter.copyright}
            </div>
            <ul className={styles["footer__nav"]}>
              {props.footerModule.bottomFooter.items.map((link, key) => (
                <li key={key}>
                  <LinkComponent to={link.url}>{link.title}</LinkComponent>
                </li>
              ))}
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};

export { FooterComponent };
