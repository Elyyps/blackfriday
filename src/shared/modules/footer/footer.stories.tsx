import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { FooterComponent } from "../footer";
import { generateFooterDataBelgian } from "@app/api/modules/footer/endpoints";

import { withProvider } from "@app/util";

storiesOf("Footer", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <FooterComponent footerModule={generateFooterDataBelgian()} />);
