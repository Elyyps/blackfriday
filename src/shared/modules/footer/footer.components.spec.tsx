import React from "react";
import { shallow } from "enzyme";
import { FooterComponent } from "./footer.component";
import { getFullfooter } from "@app/api/modules/footer/endpoints";
describe("[Footer]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<FooterComponent footerModule={getFullfooter()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
