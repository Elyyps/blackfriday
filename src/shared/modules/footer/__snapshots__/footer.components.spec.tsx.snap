// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Footer] should render component without crashing 1`] = `
<div
  className="footer"
>
  <div
    className="footer__top"
  >
    <div
      className="uk-container"
    >
      <div
        className="footer__top-holder"
      >
        <div
          className="footer__top-left"
        >
          <ul
            className="footer__top-list"
          >
            <li
              key="0"
            >
              <IconComponent
                icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/store.svg"
                size="20px"
              />
              <span
                style={
                  Object {
                    "marginLeft": "10px",
                  }
                }
              >
                132 winkels
              </span>
            </li>
            <li
              key="1"
            >
              <IconComponent
                icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/medal.svg"
                size="20px"
              />
              <span
                style={
                  Object {
                    "marginLeft": "10px",
                  }
                }
              >
                560 Deals
              </span>
            </li>
          </ul>
        </div>
        <div
          className="footer__top-right uk-visible@m"
        >
          <Connect(USP)
            uspModule={
              Object {
                "uniqueSellingPoints": Array [
                  Object {
                    "icon": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/shop.svg",
                    "link": Object {
                      "title": "Overzicht van Ã¡lle black friday winkels",
                      "url": "./stores-overview",
                    },
                  },
                  Object {
                    "icon": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/clock.svg",
                    "link": Object {
                      "title": "Altijd up-to-date van de laatste deals",
                      "url": "./deals-overview",
                    },
                  },
                  Object {
                    "icon": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/megaphone.svg",
                    "link": Object {
                      "title": "Mis niets van Black Friday 2019",
                      "url": "./",
                    },
                  },
                ],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer__holder"
  >
    <div
      className="uk-container"
    >
      <div
        className="footer__columns"
      >
        <div
          className="footer__column-collapse"
          key="0"
        >
          <div
            className="footer__column-desktop__container"
          >
            <div
              className="footer__column-title"
            >
              Wat is er nu trending
            </div>
            <div
              className="footer__column-collapse-body"
            >
              <ul
                className="footer__column-nav"
              >
                <li
                  key="0"
                >
                  <a
                    aria-label="Footbar link to Black Friday 2019"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday 2019
                  </a>
                </li>
                <li
                  key="1"
                >
                  <a
                    aria-label="Footbar link to Black Friday Apple"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Apple
                  </a>
                </li>
                <li
                  key="2"
                >
                  <a
                    aria-label="Footbar link to Black Friday Deals"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Deals
                  </a>
                </li>
                <li
                  key="3"
                >
                  <a
                    aria-label="Footbar link to Black Friday Elektronica"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Elektronica
                  </a>
                </li>
                <li
                  key="4"
                >
                  <a
                    aria-label="Footbar link to Black Friday Gaming"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Gaming
                  </a>
                </li>
                <li
                  key="5"
                >
                  <a
                    aria-label="Footbar link to Black Friday Winkels"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Winkels
                  </a>
                </li>
                <li
                  key="6"
                >
                  <a
                    aria-label="Footbar link to Cyber Monday 2019"
                    href="/"
                    rel="noopener"
                  >
                    Cyber Monday 2019
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="footer__column-mobile__container"
          >
            <ExpandablePanelComponent
              callback={[Function]}
              identifier="TRENDING_BLOCK"
              key="0"
              mainContent={
                <div
                  className="footer__column-title "
                >
                  Wat is er nu trending
                  <div
                    className="footer__column-title__icon"
                  >
                    <IconComponent
                      icon="chevron-down.svg"
                      size="10px"
                    />
                  </div>
                </div>
              }
              opened={false}
            >
              <div>
                <ul
                  className="footer__column-nav"
                >
                  <li
                    key="0"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday 2019
                    </LinkComponent>
                  </li>
                  <li
                    key="1"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Apple
                    </LinkComponent>
                  </li>
                  <li
                    key="2"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Deals
                    </LinkComponent>
                  </li>
                  <li
                    key="3"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Elektronica
                    </LinkComponent>
                  </li>
                  <li
                    key="4"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Gaming
                    </LinkComponent>
                  </li>
                  <li
                    key="5"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Winkels
                    </LinkComponent>
                  </li>
                  <li
                    key="6"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Cyber Monday 2019
                    </LinkComponent>
                  </li>
                </ul>
              </div>
            </ExpandablePanelComponent>
          </div>
        </div>
        <div
          className="footer__column-collapse"
          key="1"
        >
          <div
            className="footer__column-desktop__container"
          >
            <div
              className="footer__column-title"
            >
              Meer Black Friday
            </div>
            <div
              className="footer__column-collapse-body"
            >
              <ul
                className="footer__column-nav"
              >
                <li
                  key="0"
                >
                  <a
                    aria-label="Footbar link to Airpods Black Friday"
                    href="/"
                    rel="noopener"
                  >
                    Airpods Black Friday
                  </a>
                </li>
                <li
                  key="1"
                >
                  <a
                    aria-label="Footbar link to Black Friday Kleding"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Kleding
                  </a>
                </li>
                <li
                  key="2"
                >
                  <a
                    aria-label="Footbar link to Black Friday Jumbo"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Jumbo
                  </a>
                </li>
                <li
                  key="3"
                >
                  <a
                    aria-label="Footbar link to Black Friday Rituals"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Rituals
                  </a>
                </li>
                <li
                  key="4"
                >
                  <a
                    aria-label="Footbar link to Black Friday G-star"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday G-star
                  </a>
                </li>
                <li
                  key="5"
                >
                  <a
                    aria-label="Footbar link to Black Friday Center Parcs"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Center Parcs
                  </a>
                </li>
                <li
                  key="6"
                >
                  <a
                    aria-label="Footbar link to Black Friday Computer"
                    href="/"
                    rel="noopener"
                  >
                    Black Friday Computer
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="footer__column-mobile__container"
          >
            <ExpandablePanelComponent
              callback={[Function]}
              identifier="SHOPS_BLOCK"
              key="1"
              mainContent={
                <div
                  className="footer__column-title "
                >
                  Meer Black Friday
                  <div
                    className="footer__column-title__icon"
                  >
                    <IconComponent
                      icon="chevron-down.svg"
                      size="10px"
                    />
                  </div>
                </div>
              }
              opened={false}
            >
              <div>
                <ul
                  className="footer__column-nav"
                >
                  <li
                    key="0"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Airpods Black Friday
                    </LinkComponent>
                  </li>
                  <li
                    key="1"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Kleding
                    </LinkComponent>
                  </li>
                  <li
                    key="2"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Jumbo
                    </LinkComponent>
                  </li>
                  <li
                    key="3"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Rituals
                    </LinkComponent>
                  </li>
                  <li
                    key="4"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday G-star
                    </LinkComponent>
                  </li>
                  <li
                    key="5"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Center Parcs
                    </LinkComponent>
                  </li>
                  <li
                    key="6"
                  >
                    <LinkComponent
                      to="/"
                    >
                      Black Friday Computer
                    </LinkComponent>
                  </li>
                </ul>
              </div>
            </ExpandablePanelComponent>
          </div>
        </div>
        <div
          className="footer__column-collapse"
        >
          <div
            className="footer__column-desktop__container"
          >
            <div
              className="footer__column-title"
            >
              Eenvoudig met ons in contact
            </div>
            <div
              className="footer__column-collapse-body"
            >
              <div
                className="footer__social"
              >
                <div
                  className="footer__social-items"
                  key="0"
                >
                  <a
                    aria-label="Logo link"
                    className="footer__social-link"
                    href="https://www.facebook.com/BlackFridayNederland/"
                    rel="noopener"
                    target="_blank"
                  >
                    <div
                      className="footer__social-icon"
                    >
                      <IconComponent
                        icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/Social-Icons/facebook.svg"
                        size="15px"
                      />
                    </div>
                    <div
                      className="footer__social-text"
                    >
                      <div
                        className="footer__social-title"
                      >
                        Facebook
                      </div>
                      4.240 mensen vinden ons leuk
                    </div>
                  </a>
                </div>
                <div
                  className="footer__social-items"
                  key="1"
                >
                  <a
                    aria-label="Logo link"
                    className="footer__social-link"
                    href="https://www.instagram.com/blackfridaynl/"
                    rel="noopener"
                    target="_blank"
                  >
                    <div
                      className="footer__social-icon"
                    >
                      <IconComponent
                        icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/Social-Icons/instagram.svg"
                        size="15px"
                      />
                    </div>
                    <div
                      className="footer__social-text"
                    >
                      <div
                        className="footer__social-title"
                      >
                        Instagram
                      </div>
                      Volg ons voor leuke acties
                    </div>
                  </a>
                </div>
                <div
                  className="footer__social-items"
                  key="2"
                >
                  <a
                    aria-label="Logo link"
                    className="footer__social-link"
                    href="mailto:info@blackfridaynederland.nl"
                    rel="noopener"
                    target="_blank"
                  >
                    <div
                      className="footer__social-icon"
                    >
                      <IconComponent
                        icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/Social-Icons/mail.svg"
                        size="15px"
                      />
                    </div>
                    <div
                      className="footer__social-text"
                    >
                      <div
                        className="footer__social-title"
                      >
                        Mailtje
                      </div>
                      Heeft u een vraag? Stel hem gerust!
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="footer__column-mobile__container"
          >
            <ExpandablePanelComponent
              callback={[Function]}
              identifier="SOCIAL_BLOCK"
              mainContent={
                <div
                  className="footer__column-title "
                >
                  Eenvoudig met ons in contact
                  <div
                    className="footer__column-title__icon"
                  >
                    <IconComponent
                      icon="chevron-down.svg"
                      size="10px"
                    />
                  </div>
                </div>
              }
              opened={false}
            >
              <div
                className="footer__social"
              >
                <div
                  className="footer__social-items"
                  key="0"
                >
                  <a
                    className="footer__social-link"
                    href="https://www.facebook.com/BlackFridayNederland/"
                    rel="noopener"
                    target="_blank"
                  >
                    <div
                      className="footer__social-icon"
                    >
                      <IconComponent
                        icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/Social-Icons/facebook.svg"
                        size="15px"
                      />
                    </div>
                    <div
                      className="footer__social-text"
                    >
                      <div
                        className="footer__social-title"
                      >
                        Facebook
                      </div>
                      4.240 mensen vinden ons leuk
                    </div>
                  </a>
                </div>
                <div
                  className="footer__social-items"
                  key="1"
                >
                  <a
                    className="footer__social-link"
                    href="https://www.instagram.com/blackfridaynl/"
                    rel="noopener"
                    target="_blank"
                  >
                    <div
                      className="footer__social-icon"
                    >
                      <IconComponent
                        icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/Social-Icons/instagram.svg"
                        size="15px"
                      />
                    </div>
                    <div
                      className="footer__social-text"
                    >
                      <div
                        className="footer__social-title"
                      >
                        Instagram
                      </div>
                      Volg ons voor leuke acties
                    </div>
                  </a>
                </div>
                <div
                  className="footer__social-items"
                  key="2"
                >
                  <a
                    className="footer__social-link"
                    href="mailto:info@blackfridaynederland.nl"
                    rel="noopener"
                    target="_blank"
                  >
                    <div
                      className="footer__social-icon"
                    >
                      <IconComponent
                        icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/Social-Icons/mail.svg"
                        size="15px"
                      />
                    </div>
                    <div
                      className="footer__social-text"
                    >
                      <div
                        className="footer__social-title"
                      >
                        Mailtje
                      </div>
                      Heeft u een vraag? Stel hem gerust!
                    </div>
                  </a>
                </div>
              </div>
            </ExpandablePanelComponent>
          </div>
        </div>
        <div>
          <div
            className="footer__column-body"
          >
            <div
              className="footer__column-title"
            >
              Aanmelden voor onze nieuwsbrief
            </div>
            <p>
               
              En ontvang ons de allerbeste aanbiedingen
            </p>
            <ModalComponent
              title="Inschrijven nieuwsbrief"
              trigger={
                <Button
                  icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/newsletter/paper.svg"
                  title="Nieuwsbrief"
                  variant="primary-inverted"
                />
              }
              variant="small"
            >
              <WithFormik(InnerForm)
                onSubmit={[Function]}
              />
            </ModalComponent>
          </div>
          <div
            className="footer__column-footer"
          >
            <div
              className="rating"
            >
              <div
                className="rating__mark"
              >
                4.5
              </div>
              <StarsRatingComponent
                numberOfStars={5}
                rating={4.5}
                voteTrigger={[Function]}
              />
              <div
                style={
                  Object {
                    "marginLeft": "10px",
                  }
                }
              >
                <LinkComponent
                  to="https://www.google.com/search?output=search&q=Black+Friday+Nederland&ludocid=1020966209707573937&kgs=5565f77d0465e24f&shndl=-1&source=sh/x/kp/local&entrypoint=sh/x/kp/local#lrd=0x47c6eebed8f7cb0f:0xe2b335bc14c02b1,1"
                >
                  8 Google Reviews
                </LinkComponent>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer__bottom"
  >
    <div
      className="uk-container"
    >
      <div
        className="footer__bottom-holder"
      >
        <div
          className="footer__logo-box"
        >
          <Link
            aria-label="Logo link"
            className="uk-visible@m footer__logo"
            to="./"
          >
            <ImageComponent
              src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/logo-blackfriday.svg"
            />
          </Link>
          <span>
            Â© 2019 Â· Black Friday Expert is een initiatief van Maddox Media B.V
          </span>
        </div>
        <ul
          className="footer__nav"
        >
          <li
            key="0"
          >
            <LinkComponent
              to="/"
            >
              Partner worden?
            </LinkComponent>
          </li>
          <li
            key="1"
          >
            <LinkComponent
              to="/"
            >
              Algemene voorwaarden
            </LinkComponent>
          </li>
          <li
            key="2"
          >
            <LinkComponent
              to="/"
            >
              Privacy
            </LinkComponent>
          </li>
          <li
            key="3"
          >
            <LinkComponent
              to="/"
            >
              Cookiebeleid
            </LinkComponent>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
